meetings/2024-04/april-08.md

```diff
@@ -0,0 +1,1049 @@
+
+PFC: Thanks. That’s good information. Thank you very much.
+
+SYG: Yeah. Some more color to the complexity reduction request. This is not – I don’t think we have done a super thorough job reviewing, myself certainly not, being a non-expert in the space to call out things to cut. The user customization is an easy thing for me to point out as a possibility, given that I am not familiar with the space. And it seems like a much more niche use case to customize aspects of your daytime and calendar handling. If the champions are willing to reduce complexity there, V8 and Chrome will take any reduction in complexity that we can get. There is the code size concern and I want to thank you Philip for doing the investigative prototyping there. But there is also just the ongoing maintenance concern that here is the thing that is very likely to get written once and then let go, and that V8 is maintaining the libraries in perpetuity. And the fewer knobs it has the better chances to be well maintained in the future. It’s a pretty broad, high-level concern. And given its size today, any kind of reduction in code complexity, in code size is welcome.
+
+PFC: Okay, thanks. That’s good information as well.
+
+PFC: All right. (Slide 9) I would like to request consensus on this pull request linked here that fixes the rounding bug that I described.
+
+USA: All right. Let’s give it a minute. I would also reiterate that any sort of statements of explicit support are also welcome. All right. There doesn’t seem to be anything in the queue. So you have consensus.
+
+PFC: Okay. Thanks. (Slide 10) I took the liberty of writing a proposed summary for the note, which I will show here and paste into the notes.
+
+### Speaker's Summary of Key Points
+
+- Consensus was reached on a normative change to fix a bug in rounding that occurred in rare cases having to do with DST.
+- Over the next few weeks, we plan to dig into remaining concerns from TC39 delegates, particularly with the goal of reducing complexity.
+- Follow the checklist in [#2628](https://github.com/tc39/proposal-temporal/issues/2628) for updates.
+
+## Duplicate named capture groups for stage 4
+
+Presenter: Kevin Gibbons (KG)
+
+- [PR](https://github.com/tc39/ecma262/pull/2721)
+
+KG: Okay. So duplicate named capture groups. As a reminder, since it’s been a couple of years since I presented this, this is a feature that allows you to have the same capturing group name into two parts of a regular expression, with the constraint that they can’t both participate in the match, which is to say they have to be in different alternatives. So separated by a pipe. But otherwise, it works exactly like you would expect. You can use back references to the capturing group name. The.groups object of the result will contain the value from whichever one actually matched. In the case of repetition groups the last reputation group defines it, the same way it works for regular capturing groups.
```

Comment:
```suggestion
KG: Okay. So duplicate named capture groups. As a reminder, since it’s been a couple of years since I presented this, this is a feature that allows you to have the same capturing group name into two parts of a regular expression, with the constraint that they can’t both participate in the match, which is to say they have to be in different alternatives. So separated by a pipe. But otherwise, it works exactly like you would expect. You can use back references to the capturing group name. The.groups object of the result will contain the value from whichever one actually matched. In the case of repetition groups the last repetition group defines it, the same way it works for regular capturing groups.
```